### ISSUE TITLE
iina doesn't close gifs fully when moving to the next playlist item

### ISSUE THREAD

**Issue by ashleyharvey** [iina/iina#2580]
- macOS 10.14.6
- IINA 1.0.4 (107)

**Expected behavior:**
I use `lsof` to monitor which file(s) iina has open in my scripts so that I can tag/move/delete/organize media effectively.  It works well until I encounter gifs.  iina seems to open new file handles and not necessarily close them the same way it does with other media.  Once the file handles start stacking up, my script(s) have a hard time differentiating the currently-being-read file from the others.  Especially since sometimes it seems like file handles get re-used and other times not, so I can't easily just go to the top or bottom of the stack and have any guarantee that that's the most recently played media.

**Actual behavior:**
File handles should be closed (or not) when advancing to the next playlist item.  It shouldn't be that some medias file handles are closed and some not; it should be consistent either way.

<details>
<summary><b>Crash report:</b></summary>

```
```
</details>

<br>

<details>
<summary><b>mpv log:</b></summary>

```
```
</details>

<br>

**Steps to reproduce:**

- [√] MPV does not have this problem.

**How often does this happen?**
It's easily reproducible.  Create a playlist with video and gif files, use `lsof -r` to monitor the process it, and you'll see it opening (and not closing) the gifs.  The other formats seem to behave as expected.

**Comment by ashleyharvey** [comment]
Update on this issue: I enabled logging and tried to see if there was a difference between files being opened and closed properly and not. As it turns out, *if thumbnails are successfully generated,* then it seems like the file is closed properly. I confirmed by playing only gifs and playing only other media and checking the logs. Turning off the thumbnail caching feature has worked around the problem.
